netbox/vpn/forms/filtersets.py

```diff
@@ -0,0 +1,182 @@
+from django import forms
+from django.utils.translation import gettext as _
+
+from netbox.forms import NetBoxModelFilterSetForm
+from tenancy.forms import TenancyFilterForm
+from utilities.forms.fields import DynamicModelMultipleChoiceField, TagFilterField
+from vpn.choices import *
+from vpn.models import *
+
+__all__ = (
+    'IKEPolicyFilterForm',
+    'IKEProposalFilterForm',
+    'IPSecPolicyFilterForm',
+    'IPSecProfileFilterForm',
+    'IPSecProposalFilterForm',
+    'TunnelFilterForm',
+    'TunnelTerminationFilterForm',
+)
+
+
+class TunnelFilterForm(TenancyFilterForm, NetBoxModelFilterSetForm):
+    model = Tunnel
+    fieldsets = (
+        (None, ('q', 'filter_id', 'tag')),
+        (_('Tunnel'), ('status', 'encapsulation', 'tunnel_id')),
+        (_('Security'), ('ipsec_profile_id',)),
+        (_('Tenancy'), ('tenant_group_id', 'tenant_id')),
+    )
+    status = forms.MultipleChoiceField(
+        label=_('Status'),
+        choices=TunnelStatusChoices,
+        required=False
+    )
+    encapsulation = forms.MultipleChoiceField(
+        label=_('Encapsulation'),
+        choices=TunnelEncapsulationChoices,
+        required=False
+    )
+    ipsec_profile_id = DynamicModelMultipleChoiceField(
+        queryset=IPSecProfile.objects.all(),
+        required=False,
+        label=_('IPSec profile')
+    )
+    tunnel_id = forms.IntegerField(
+        required=False,
+        label=_('Tunnel ID')
+    )
+    tag = TagFilterField(model)
+
+
+class TunnelTerminationFilterForm(NetBoxModelFilterSetForm):
+    model = TunnelTermination
+    fieldsets = (
+        (None, ('q', 'filter_id', 'tag')),
+        (_('Termination'), ('tunnel_id', 'role')),
+    )
+    tunnel_id = DynamicModelMultipleChoiceField(
+        queryset=Tunnel.objects.all(),
+        required=False,
+        label=_('Tunnel')
+    )
+    role = forms.MultipleChoiceField(
+        label=_('Role'),
+        choices=TunnelTerminationRoleChoices,
+        required=False
+    )
+    tag = TagFilterField(model)
```

Comment:
Add type? (and device?)

Reply:
I've added a filter for `termination_type`. Related objects may be a bit more involved; going to punt on those for now.